src/librustc_lint/builtin.rs

```diff
@@ -626,9 +625,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for MissingDebugImplementations {
 
 declare_lint! {
     pub ANONYMOUS_PARAMETERS,
-    Allow,
-    "detects anonymous parameters",
-    Edition::Edition2018 => Warn
+    Warn,
+    "detects anonymous parameters"
```

Comment:
So, I think that the policy in *general* is that we only warn in migration mode -- but this is an unconditional warning, no? I don't really care either way, I just want to Do What's Right.

cc @Mark-Simulacrum @Centril

Reply:
We talked about this a bit yesterday on Discord; I think our conclusion was that warnings only during migration was the policy but that in this case an unconditional warning would be OK? Did you discuss this further?

Reply:
I believe if we just wanted to warn in migration mode, we would change this back to `Allow` and add it to the `FutureCompatibilityLints` array. I don't really see why it should be different from the other migration lints, but I don't have a strong preference.

Reply:
Yes, but I need to learn how to block the syntax in the new edition, whereas I already know how to change the lint. I was thinking that after this PR, we could submit another that removes the syntax in 2018 edition (I will need some help with that). In the mean time, we can at least start warning as early as possible, right?

Reply:
we could of course always do that in a follow-up PR...

Reply:
Anyway I basically don't care if we warn in 2015 always or only when migrating. I thought the typical policy was only the latter, though. The only reason to do otherwise would be to give a kind of "extra nudge" -- the fact that we accepted a separate RFC (pre-edition) which deprecated this syntax might be enough justification. This would be analogous to deprecating some API, which obviously doesn't have to occur on an edition boundary.

Reply:
Ok, I opened #53612. This might actually be a lot easier than I expected.